syncer/syncer.go

```diff
@@ -863,25 +869,42 @@ func (s *Syncer) updateReplicationLag(job *job, lagKey string) {
 				time.Sleep(time.Second * time.Duration(t))
 			}
 		})
-
-		switch job.tp {
-		case ddl, skip:
-			// NOTE: we handle ddl/skip job separately because ddl job will clean all the dml job before execution
-			lag = s.calcReplicationLag(int64(job.eventHeader.Timestamp))
-		default: // dml job
-			// NOTE workerlagmap already init all dml key(queueBucketName) before syncer running.
-			s.workerLagMap[lagKey].Store(s.calcReplicationLag(int64(job.eventHeader.Timestamp)))
-		}
+		lag = s.calcReplicationLag(int64(job.eventHeader.Timestamp))
+		s.workerLagMap[lagIdx].Store(lag)
 	}
+}
+
+func (s *Syncer) updateReplicationLagMetric() {
+	var lag int64
 	// find all job queue lag choose the max one
-	for _, l := range s.workerLagMap {
-		if wl := l.Load(); wl > lag {
+	for idx := range s.workerLagMap {
+		if wl := s.workerLagMap[idx].Load(); wl > lag {
 			lag = wl
 		}
 	}
 	metrics.ReplicationLagHistogram.WithLabelValues(s.cfg.Name, s.cfg.SourceID, s.cfg.WorkerName).Observe(float64(lag))
 	metrics.ReplicationLagGauge.WithLabelValues(s.cfg.Name, s.cfg.SourceID, s.cfg.WorkerName).Set(float64(lag))
 	s.secondsBehindMaster.Store(lag)
+
+	if lag > 1 {
+		s.tctx.L().Warn("current replication lag greater than 1(s)",
```

Comment:
this maybe cause too much log when lag is alway large than 1 ? and this may cause high IO pressure 🤔

Reply:
better to remove it because if there're many backlog binlog files it will print many log